department-of-veterans-affairs/caseflow
# Ki/APPEALS-35714 - Update Queue shell to have a Team Queue version

Resolves [APPEALS-35714 - Update Queue shell to have a Team Queue version](https://jira.devops.va.gov/browse/APPEALS-35714)

# Description
- Updated correspondence_controller to replace correspondence_cases with new call for the specific users assigned correspondence tasks
- Created new correspondence_team call to populate _all_ correspondence tasks if user is MailTeamSupervisor
- Updated routing for queue/correspondence/team and related .erb
- Updated correspondence routing in QueueApp for the team and own cases
- Slight updates to CorresopndenceCases.jsx and CorrespondenceTable.jsx for getting correspondenceTasks to load in instead of vetCorrespondences (for testing and future work)
NOTE: Updates to correspondence_controller_spec.rb still need to be made, but need to verify future work as it's all written with vet/correspondence work hanging off of it.

## Acceptance Criteria

## Testing Plan
This is a back-end story. so front-end components will register as "undefined" until the additional details and options are built out
1. User Test
    1. Log in as JOLLY_POSTMAN
    2. Navigate to queue/correspondence/team
    3. Validate that it redirects to queue/correspondence/
    4. Make sure you're on queue/correspondence/
    5. Validate correspondenceConfig populates table/task information
2. Team/Supervisor Test
    1. Log in as MAIL_TEAM_SUPERVISOR_MAIL_INTAKE_USER
    2. Navigate to queue/correspondence/team/
    3. Validate correspondenceConfig populates table/task information

### All correspondenceTasks getting passed in [OUTDATED]
![Screenshot 2024-01-19 at 3 40 30 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/108023343/456f7df2-0198-4250-ad4b-bbe77a2d5022)

### New data for team from correspondenceConfig
![Screenshot 2024-01-30 at 12 13 19 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/108023343/f65d57ce-530b-448a-bfe0-ea2b0b053b82)

### Just correspondenceTasks assigned to user passed in (JOLLY_POSTMAN) [OUTDATED]
![Screenshot 2024-01-19 at 3 03 37 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/108023343/1e696e99-dda0-4ecf-ad42-f11d04d7652e)

### New data for user from correspondenceConfig
![Screenshot 2024-01-30 at 12 12 22 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/108023343/b2dda831-53b6-447d-8760-379fdf5b9be5)


# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:
